src/Configuration/Config.EnvironmentVariables/test/EnvironmentVariablesTest.cs

```diff
@@ -85,23 +85,24 @@ public void LoadKeyValuePairsFromAzureEnvironmentWithPrefix()
         {
             var dict = new Hashtable()
             {
-                {"CUSTOMCONNSTR_db1", "CustomConnStr"},
-                {"SQLCONNSTR_db2", "SQLConnStr"},
-                {"MYSQLCONNSTR_db3", "MySQLConnStr"},
-                {"SQLAZURECONNSTR_db4", "SQLAzureConnStr"},
+                {"CUSTOMCONNSTR_PREFIX_db1", "CustomConnStr"},
+                {"SQLCONNSTR_PREFIX_db2", "SQLConnStr"},
+                {"MYSQLCONNSTR_PREFIX_db3", "MySQLConnStr"},
+                {"SQLAZURECONNSTR_PREFIX_db4", "SQLAzureConnStr"},
                 {"CommonEnv", "CommonEnvValue"},
             };
-            var envConfigSrc = new EnvironmentVariablesConfigurationProvider("ConnectionStrings:");
+            var envConfigSrc = new EnvironmentVariablesConfigurationProvider("PREFIX_");
 
             envConfigSrc.Load(dict);
 
-            Assert.Equal("CustomConnStr", envConfigSrc.Get("db1"));
-            Assert.Equal("SQLConnStr", envConfigSrc.Get("db2"));
-            Assert.Equal("System.Data.SqlClient", envConfigSrc.Get("db2_ProviderName"));
-            Assert.Equal("MySQLConnStr", envConfigSrc.Get("db3"));
-            Assert.Equal("MySql.Data.MySqlClient", envConfigSrc.Get("db3_ProviderName"));
-            Assert.Equal("SQLAzureConnStr", envConfigSrc.Get("db4"));
-            Assert.Equal("System.Data.SqlClient", envConfigSrc.Get("db4_ProviderName"));
+            Assert.Equal("CustomConnStr", envConfigSrc.Get("ConnectionStrings:db1"));
```

Comment:
Can we leave existing tests alone and add a new test for the new behavior?

Reply:
@HaoK This test as it currently stands breaks with the changes

Reply:
Right which makes this change an regression then

Reply:
Yes, it would be a regression. Do you know why anyone wanted this behaviour in the first place? Supporting this one weird case (unwrapping a magically-mapped Azure ConnectionString) is what's caused every other prefix to be completely broken.